### PR TITLE
Combobox: Support auto scrolling for combobox keyboard navigation

### PR DESCRIPTION
**Pull request by vovakulikov**
## Problem 
Prior to this PR `Combobox` component didn't scroll the suggestion list if user navigation goes through suggested items that are hidden due to list scroll sizes. See the following video with this problem.

https://user-images.githubusercontent.com/18492575/194866437-fb9aabbf-47ef-483b-ba85-31b730cf4c21.mov

You can see that the last item is partially hidden due to list sizes; when we navigate to this item, it doesn't show up in the visible area. It makes hidden items in the list scroll container unreachable with the keyboard.

https://user-images.githubusercontent.com/18492575/194868638-0276e925-424e-4dab-9124-ba7000e09a69.mov

## Test plan
- Check that it's possible to see all items in the suggested item list even if they are hidden (as you navigate to them with the keyboard)

## App preview:

- [Web](https://sg-web-vk-fix-combobox-navigation.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-nabxazijgg.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/wildcard/src/components/Combobox/Combobox.tsx [hunk 1]
```diff
@@ -171,10 +227,52 @@ export const ComboboxOption = forwardRef((props, ref) => {
...
+ * ┃  │                     │  ┃             │                     │
+ *    │                     │             ┏Container viewport ━ ━ ━│━ ┓
+ * ┃  │┌ Element ─ ─ ─ ─ ─ ┐│  ┃ ──────▶     │┌ Element ─ ─ ─ ─ ─ ┐│
+ *    │ ░░░░░░░░░░░░░░░░░░░ │             ┃  │ ░░░░░░░░░░░░░░░░░░░ │  ┃
+ * ┗ ━│╋░━░━░━░━░━░━░━░━░━░╋│━ ┛             ││░░░░░░░░░░░░░░░░░░░││
+ *    │ ░░░░░░░░░░░░░░░░░░░ │             ┃  │ ░░░░░░░░░░░░░░░░░░░ │  ┃
+ *    │└ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘│                │└ ─ ─ ─ ─ ─ ─ ─ ─ ─ ┘│
+ *    └─────────────────────┘             ┗ ━└━─━─━─━─━─━─━─━─━─━─━┘━ ┛
+ *```
+ */
```
Comment (eseliger) [inline client/wildcard/src/components/Combobox/Combobox.tsx:254]: wow, big bonus points for this piece of art :)

**Review by eseliger — APPROVED** [review]
Code looks legit, nice improvement 🔥

**Review by limitedmage — APPROVED** [review]
Very nice!

**Review by vdavid — APPROVED** [review]
Nice! :)

**Comment by vovakulikov** [comment]
I've noticed that we have some pickers that don't use Combobox UI, but it seems they should. The most popular component like this is ConnectionForm. @umpox, I see that you wrote them. Let's have a sync, I think the connection form will benefit from the new Combobox UI a lot.

**Comment by umpox** [comment]
@vovakulikov Agree, I've put something in your calendar for when you're back!

**Comment by vdavid** [comment]
@vovakulikov if you think it makes sense for a wider audience of front-end devs to learn the things you mean to tell @umpox about the Combobox UI (stuff that's not specific to the ConnectionForm) then I suggest you create a video of your thoughts and share it, maybe even add it to our [dev docs](https://docs.sourcegraph.com/dev/how-to).
I'd be personally interested because I'll need to refactor an `<input>` for the "Open in Editor" feature to add a checkbox next to each item (https://github.com/sourcegraph/sourcegraph/issues/41427).
But I guess others might find it helpful as well.

**Pull request merged**